cpp/src/IceStorm/Scanner.l

```diff
@@ -1,64 +1,76 @@
+#include "IceUtil/ScannerConfig.h"
+#include <cstdint>
+
+}
+
+%{
+
 #include "Ice/Ice.h"
 #include "Parser.h"
 #include "Grammar.h"
 
-#if defined(_MSC_VER)
-// '<' : signed/unsigned mismatch
-#   pragma warning(disable:4018)
-// 'initializing' : conversion from '__int64' to 'int', possible loss of data
-#   pragma warning(disable:4244)
-
-//
-// '=' : conversion from 'size_t' to 'int', possible loss of data
-// The result of fread() is a size_t and gets inserted into an int
-//
-#   pragma warning(disable:4267)
-#endif
-
-#if defined(__GNUC__)
-#   pragma GCC diagnostic ignored "-Wsign-compare"
+#ifdef _MSC_VER
+#    ifdef yywrap
+#        undef yywrap
+#        define yywrap() 1
+#    endif
+#    define YY_NO_UNISTD_H
 #endif
 
 using namespace std;
 using namespace Ice;
 using namespace IceStorm;
 
-#ifdef _MSC_VER
-#   ifdef yywrap
-#      undef yywrap
-#      define yywrap() 1
-#   endif
-#   define YY_NO_UNISTD_H
-#endif
-
-#define YY_INPUT(buf, result, maxSize) parser->getInput(buf, result, maxSize)
-
 namespace IceStorm
 {
+    typedef map<string, int> StringTokenMap;
+    StringTokenMap keywordMap;
 
-typedef std::map<std::string, int> StringTokenMap;
-static StringTokenMap keywordMap;
-
-void initScanner();
-
+    void initScanner();
 }
-#define         YY_USER_INIT initScanner();
+
+// Override some of the functions flex auto-generates with our own implementations.
+#define YY_USER_INIT initScanner();
+#define YY_INPUT(buf, result, maxSize) parser->getInput(buf, result, maxSize)
 
 %}
 
+  /* Instructs flex to not suppress any warnings when generating the scanner. */
+%option warn
+  /* By default flex will 'default match' any text it encounters that doesn't match any specified rules. This
+   * option disables default-matching (it throws 'scanner jammed' instead) to make grammar holes more obvious. */
+%option nodefault
+  /* Directs flex to generate a scanner tailored for use by bison.
+   * These options change the signature of the main lexing function, which must match the one declared in Grammar.y */
+%option bison-bridge
+
+  /* Ensures flex generates a scanner that supports reading 8-bit characters. */
+%option 8bit
+  /* Directs flex to generate lookup tables that are better aligned in memory to
+   * improve access speeds, even if this means allocating larger tables. */
+%option align
+  /* Directs flex to store matched text as 'char *' instead of char arrays, for improved performance. */
+%option pointer
+  /* We always want the scanner to run in interactive mode. */
+%option always-interactive
+
+  /* Disables the generation of functions we don't use to reduce clutter, and possibly improve performance. */
+%option noyywrap
+%option noyy_scan_buffer noyy_scan_bytes noyy_scan_string
+%option noyyget_extra noyyset_extra noyyget_leng noyyget_text
+%option noyyget_in noyyset_in noyyget_out noyyset_out
+%option noyyget_lineno noyyset_lineno noyyget_lloc noyyset_lloc
+%option noyyget_lval noyyset_lval noyyget_debug noyyset_debug
```

Comment:
Tell Flex not to generate any functions we don't use to cut down on the number of warnings, and make things leaner.

I checked the generated code, and these functions are indeed unused. We do the same thing in:
https://github.com/zeroc-ice/ice/blob/3111cba6b2116e61b479f368f7cc68caf00862e0/cpp/src/Slice/Scanner.l#L92

Reply:
Should we add these options to the IceGrid scanner too?